### PR TITLE
feat: support client interceptors

### PR DESCRIPTION
**Pull request by shaxbee**

### CODE REVIEW

**Comment by shaxbee** [comment]
@mraleph Tests fail due to dartanalyzer complaining about double ignored rule that was not touched in this PR.

**Comment by mraleph** [comment]
Yeah, I just made a fix for it here: https://github.com/dart-lang/protobuf/pull/430

File: protoc_plugin/test/goldens/grpc_service.pbgrpc [hunk 1]
```diff
@@ -31,35 +31,32 @@ class TestClient extends $grpc.Client {
...
-    return $grpc.ResponseFuture(call);
+    return $createStreamingCall(_$clientStreaming, request, options: options)
+        .single;
   }
 
   $grpc.ResponseStream<$0.Output> serverStreaming($0.Input request,
       {$grpc.CallOptions options}) {
-    final call = $createCall(
-        _$serverStreaming, $async.Stream.fromIterable([request]),
+    return $createStreamingCall(_$serverStreaming, Stream.value(request),
```
Comment (mraleph) [inline protoc_plugin/test/goldens/grpc_service.pbgrpc:53]: `Stream.value` <- this does not match the code written above which uses `Stream.fromIterable([request])`. Needs to be regenerated?

**Review by mraleph — CHANGES_REQUESTED** [review]
LGTM. Please regenerate goldens and rebase

File: protoc_plugin/test/goldens/grpc_service.pbgrpc (hunk 1 above)
Comment (shaxbee) [reply protoc_plugin/test/goldens/grpc_service.pbgrpc]: Regenerate goldens. Nice catch!

**Comment by shaxbee** [comment]
@mraleph regenerated goldens and rebased.

**Comment by ashley-3r** [comment]
Hi @mraleph, any more changes required ? Let us know.

**Review by mraleph — APPROVED** [review]

**Pull request merged**